From: Aikar <aikar@aikar.co>
Date: Tue, 5 Apr 2016 21:38:58 -0400
Subject: [PATCH] Remove Debug checks from DataBits

These are super hot and causing noticeable hits

Before: http://i.imgur.com/nQsMzAE.png
After: http://i.imgur.com/nJ46crB.png

diff --git a/src/main/java/net/minecraft/server/DataBits.java b/src/main/java/net/minecraft/server/DataBits.java
--- a/src/main/java/net/minecraft/server/DataBits.java
+++ b/src/main/java/net/minecraft/server/DataBits.java
@@ -13,7 +13,7 @@ public class DataBits {
     }
 
     public DataBits(int i, int j, long[] along) {
-        Validate.inclusiveBetween(1L, 32L, (long)i);
+        //Validate.inclusiveBetween(1L, 32L, (long)i); // Paper
         this.d = j;
         this.b = i;
         this.a = along;
@@ -25,8 +25,8 @@ public class DataBits {
     }
 
     public void a(int i, int j) {
-        Validate.inclusiveBetween(0L, (long)(this.d - 1), (long)i);
-        Validate.inclusiveBetween(0L, this.c, (long)j);
+        //Validate.inclusiveBetween(0L, (long)(this.d - 1), (long)i); // Paper
+        //Validate.inclusiveBetween(0L, this.c, (long)j); // Paper
         int k = i * this.b;
         int l = k / 64;
         int i1 = ((i + 1) * this.b - 1) / 64;
@@ -41,7 +41,7 @@ public class DataBits {
     }
 
     public int a(int i) {
-        Validate.inclusiveBetween(0L, (long)(this.d - 1), (long)i);
+        //Validate.inclusiveBetween(0L, (long) (this.d - 1), (long) i); // Paper
         int j = i * this.b;
         int k = j / 64;
         int l = ((i + 1) * this.b - 1) / 64;
-- 
2.19.1

